### DISCUSSION TITLE
[答疑] 为什么NapCat会更新如此快? 看上去如此不稳定呢?

### DISCUSSION THREAD

**Discussion by MliKiowa** [NapNeko/NapCatQQ#517] [General]
事实上NapCatQQ是当前 非协议 实现最为复杂与稳定的.

## napcat为什么如此频繁更新
由于napcat docker用户居多 并没有内置更新功能 为了进行测试与立刻到达用户 napcat每修一个问题就会迭代一个版本号

除此还因为之前action脚本并没有为commit打包 以便测试

## 为什么频繁变更启动方式呢

在之前NTQQ定点的干掉了napcat实现的一处原理

所谓你若不动一切安好 你若要动那么反而就要迎上去了

于是napcat启动了3+数目的方案以保证napcat的正常使用 在此期间 各种方案的优劣也有所观察 由此napcat进行了一次大型重构

## 最稳定的非协议bot框架?
是的 就是napcat
 
1. 无关上游框架 napcat的framework版不只是ll插件 它可以独立运行在QQ的main进程 
 可以迁移任何框架 也就是说框架的维护状态无关napcat 甚至napcat framework本身就是框架
3. 如果再次定点打击napcat的技术呢? 这是不用担心的 基于最新的研究与能力的提高 napcat已初步具有全平台特征分析与hook能力 对major与wrapper研究以处于极为领先的地位 新的启动器正在来临(包括当前技术原理的与真无头技术的)
4. napcat在实现上不同于其余框架 生态链与周边技术 几乎100％自有
启动器 图形化工具 自动化汇编分析 HookJs 反检测frida脚本 hook node模块 等工具的技术拥有
5. 无关renderer 由于electron前端ipc交互更为复杂 变化更为频繁 其接口稳定性与napcat位于的层级相比更为不稳定
6. 绕不开napcat的技术原理 napcat早起实践于 msgid lru/消息发送队列 对基础处理更为深入 uin/uid转换也是经过长期测试与考虑
7. 接口与标准 napcat现已完成99％的onebot和gocq接口实现 并对部分接口扩展 远超大部分框架 具有独立rkey获取能力

**Comment by Funny-ppt** [comment]
现在是不是又被定点打击了, 一周前左右突然正常登陆的账号被检测为异常并踹下线, 然后提醒我可能使用了第三方客户端面临封禁(不过我更新不是很勤快, 部署之后基本没有很经常pull过)

**Reply by KaedeharaLu** [reply]
可能是，更新还是有必要的，遇到这种问题可以更新一下，换个号，过段时间再换回来（亲身经历）

**Comment by 837951602** [comment]
由於在本地做了點小修改，不是很方便更新。目前用了四個月，剛開始封過一次，後來也沒什麼問題（包括剛開始用的llonebot，後來轉napcat的）
